### ISSUE TITLE
bug: ImmutableArray.Contains used without SymbolEqualityComparer in 2 analyzers

### ISSUE THREAD

**Issue by rjmurillo** [rjmurillo/moq.analyzers#973]
## Summary

Two analyzers use `ImmutableArray<IMethodSymbol>.Contains()` which uses default reference equality instead of `SymbolEqualityComparer`. This can cause false negatives where valid Moq method calls are not detected.

### RaiseEventArgumentsShouldMatchEventSignatureAnalyzer

**File:** `src/Analyzers/RaiseEventArgumentsShouldMatchEventSignatureAnalyzer.cs`, line 141

```csharp
return knownSymbols.Mock1Raise.Contains(methodSymbol.OriginalDefinition);
```

### EventSetupHandlerShouldMatchEventTypeAnalyzer

**File:** `src/Analyzers/EventSetupHandlerShouldMatchEventTypeAnalyzer.cs`, line 164

```csharp
knownSymbols.ItIsAny.Contains(methodSymbol)
```

## Why This Is a Bug

`ImmutableArray<T>.Contains` uses `EqualityComparer<T>.Default`, which for `ISymbol` is reference equality. Roslyn does not guarantee reference equality for symbols. The rest of the codebase correctly uses `IsInstanceOf(knownSymbols.Xxx)` which uses `SymbolEqualityComparer.Default`.

The project instructions explicitly mandate: "Use `ISymbol`, `SymbolEqualityComparer`, and `MoqKnownSymbols` for all type resolution."

## Impact

- **False negatives**: Moq Raise/It.IsAny calls may not be detected, leading to missed diagnostics
- **Non-deterministic**: May work in some compilations and fail in others depending on symbol identity

## Suggested Fix

Use `methodSymbol.IsInstanceOf(knownSymbols.Mock1Raise)` or `knownSymbols.Mock1Raise.Any(m => SymbolEqualityComparer.Default.Equals(m, methodSymbol.OriginalDefinition))` for consistency with the rest of the codebase.

## Discovered by

Automated codebase investigation (Code Review agent).
